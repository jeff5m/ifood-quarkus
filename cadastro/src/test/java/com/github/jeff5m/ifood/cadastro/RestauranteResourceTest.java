package com.github.jeff5m.ifood.cadastro;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static io.restassured.RestAssured.given;

@QuarkusTest
@QuarkusTestResource(CadastroTestLifeCycleManager.class)
@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
class RestauranteResourceTest {

    @Test
    @DisplayName("list returns list of restaurants when successful")
    @DataSet("restaurants-scenario-1.yml")
    void list_ReturnsListOfRestaurants_WhenSuccessful() {
        String result = given()
                .when().get("/restaurantes")
                .then()
                .statusCode(200)
                .extract().asString();
        Approvals.verifyJson(result); // change json generated file from "*.received.json" to "*.approved.json"
    }
}
