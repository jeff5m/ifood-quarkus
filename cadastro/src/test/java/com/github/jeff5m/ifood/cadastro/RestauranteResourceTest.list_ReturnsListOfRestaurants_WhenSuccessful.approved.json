[
  {
    "id": 123,
    "nome": "Novo Restaurante",
    "proprietario": "ID do keyclok"
  },
  {
    "id": 456,
    "nome": "Novo Restaurante 2",
    "proprietario": "ID do keyclok 2"
  }
]